packages/core/src/libraries/hook/index.test.ts

```diff
@@ -50,7 +49,7 @@ const findAllHooks = jest.fn().mockResolvedValue([hook]);
 const findHookById = jest.fn().mockResolvedValue(hook);
 
 const { createHookLibrary } = await import('./index.js');
-const { triggerInteractionHooksIfNeeded, attachExecutionStatsToHook, testHook } = createHookLibrary(
+const { triggerInteractionHooks, attachExecutionStatsToHook, testHook } = createHookLibrary(
```

Comment:
nit: `attachExecutionStatsToHook` attach is not precise IMHO. maybe `updateExecutionStats`

Reply:
`attachExecutionStatsToHook` is used to attach execution stats to hook data; it's not about updating.